### ISSUE TITLE
Restarting a training is not following the lr pattern laid out in multiple-step

### ISSUE THREAD

**Issue by stickzh** [DeepLabCut/DeepLabCut#1646]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

When stopping and restarting a training from a snapshot the match-step starts from the first step definition. However because it requires an exact match between the iteration number and the parameter in match-step it can be stuck in the fist definition for a surprising time!
For example:
'multi_step': [[0.005, 10000], [0.02, 430000], [0.002, 730000], [0.001, 1030000]],
if run  continuously, lr would be .05 till 10000, then .02 till 430000, then etc etc.
If however you stop at iteration 50000, and restart from that snapshot:
lr will start at .005, but since iterations will never be at 10000 again, lr will stay at .005 until iteration 430000 when it will jump to .002.
By changing the multi-step between restarts then one can overcome this, bit it took a while before I understood what was happening

I think the "problem" is in the code below which requires an exit match of iteration number to move to the next lr value:

> class LearningRate(object):
>     def __init__(self, cfg):
>         self.steps = cfg["multi_step"]
>         self.current_step = 0
> 
>     def get_lr(self, iteration):
>         lr = self.steps[self.current_step][0]
>         if iteration == self.steps[self.current_step][1]:
>             self.current_step += 1
> 
>         return lr

### Operating System

Mac-m1

### DeepLabCut version

2.2.0.3

### DeepLabCut mode

single animal

### Device type

gpu
Mac M1

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by stickzh** [comment]
I am also by Geneva (CERN) if you need any further info

**Comment by stickzh** [comment]
Thank you for the fix and the clarification about max-iter

**Comment by Tetra-quark** [comment]
I have a follow-up to this and its fix #1653. Restarting training now reads the `multi_step` parameter in a relative way meaning that if restarted at 50k iterations with:
```
'multi_step': 
[[0.005, 10000], 
[0.02, 430000], 
[0.002, 730000], 
[0.001, 1030000]]
```
The first 10k iterations 50k-60k will have a learning rate of 0.005 after which it will switch to 0.02. Meanwhile the first training run will have used 0.005 from 0-10k and then 0.02 from 10k-50k. If I were to restart training in this way multiple times over it may even mean the learning rate never gets to 0.002 and 0.001. I would imagine that a learning rate schedule such as this is likely to be sub-optimal. 

This obviously isn't the most common way to train a network because under normal circumstances you would expect the model to train until the maxiters of 1,030,000 in this example. 

But this might be an issue for Colab users. In my case, we experience frequent random interruptions to training despite using Colab Pro. Training usually runs for about 4-6 hours after which a `FileNotFound` error will be raised for one of the training frames. I have speculated as to whether the mounted google drive folder times out or something but I am not sure what is ultimately at the origin of this interruption. 

The loss in tensorboard seems reasonable despite these interruptions but I just wanted to point out this observation just in case.

**Comment by jvitte** [comment]
Hi @Tetra-quark I have the exact same questions since I am also experiencing random interruptions while using Colab (using pay as you go compute units). Were you able to find answers?

**Comment by Tetra-quark** [comment]
It was quite a while ago but I don't believe I did, I moved on to other things using a model that was good enough for the time being. Might be worth opening a new issue for visibility!
